src/platform-aws.c

```diff
@@ -318,47 +319,28 @@ static int configure_ep_max_msg_size(struct fid_ep *ep)
 	return ret;
 }
 
-
-typedef ncclResult_t (*nccl_get_version_fn_t)(int *version);
-
 static int configure_nvls_option(void)
 {
 	/* Disable NVLS topology discovery for older NCCL versions. There's a
 	 * bug with EFA and NCCL version 2.18.3 and earlier on platforms with
 	 * NVLink Switch support.  We selectively disable NVLS support
 	 * to avoid the bug, which was fixed in 2.18.5.
 	 */
-	nccl_get_version_fn_t nccl_get_version = NULL;
-	int version = 0;
-	ncclResult_t nccl_ret;
 	int ret;
 
 	if (getenv("NCCL_NVLS_ENABLE") == NULL) {
-		nccl_get_version = (nccl_get_version_fn_t)dlsym(RTLD_DEFAULT, "ncclGetVersion");
-		if (nccl_get_version == NULL) {
-			NCCL_OFI_TRACE(NCCL_INIT | NCCL_NET,
-			    "Could not find ncclGetVersion symbol; skipping NVLS NCCL version check");
-			return 0;
-		} else {
-			nccl_ret = nccl_get_version(&version);
-			if (nccl_ret != ncclSuccess) {
-				NCCL_OFI_WARN("ncclGetVersion returned %d", nccl_ret);
-				return -ENOTSUP;
-			}
-
-			NCCL_OFI_TRACE(NCCL_INIT | NCCL_NET, "ncclGetVersion results = %d", version);
-		}
-
 		/* 2.18.5 */
-		if (version < 21805) {
-			NCCL_OFI_INFO(NCCL_INIT | NCCL_NET, "Disabling NVLS support due to NCCL version %d", version);
+		if (NCCL_VBEFORE(nccl_ofi_ncclver_get(), NCCL_V(2,18,5))) {
```

Comment:
Won't this resolve to true if we have unknown NCCL version? And then we'll disable NVLS when we shouldn't? That's the problem we had before.